charmdir.go

```diff
@@ -460,21 +467,21 @@ func (dir *CharmDir) MaybeGenerateVersionString(logger Logger) (string, string,
 	} else if _, err = os.Stat(filepath.Join(dir.Path, ".git")); err == nil {
 		cmdArgs = []string{"git", "describe", "--dirty", "--always"}
 	} else if file, err := os.Open(dir.join("version")); err == nil {
-		logger.Debugf("charm is not in version control, but uses a version file")
+		logger.Debugf("charm is not in version control, but uses a version file, charm path %q", dir.Path)
 		var versionNumber string
 		n, err := fmt.Fscan(file, &versionNumber)
 		if n != 1 {
-			return "", "versionFile", errors.Errorf("invalid version file")
+			return "", versionFileVersionType, errors.Errorf("invalid version file, charm path: %q", dir.Path)
 		}
 		if err != nil {
-			return "", "versionFile", err
+			return "", versionFileVersionType, err
```

Comment:
I was thinking about taking all of those "vcstype" out into global constants. Thus we could reuse them e.g. in tests and do not have to use "magical" strings. But this would warrant additional changes I wasn't ready to commit